### ISSUE TITLE
dbg! should be removed

### ISSUE THREAD

**Issue by carlomilanesi** [rust-lang/rust-clippy#9060]
### What it does

Checks that the code does not contains any call to the dbg macro, except in comments. This macro is not suited for production code.

### Lint Name

dbg_should_be_removed

### Category

correctness

### Advantage

If this macro is not removed, some debug lines may be printed to stderr.

### Drawbacks

Some developers may prefer to print debug info in error messages printed to stderr, even in production code.

### Example

```rust
dbg!(42);
```

Could be written as:

```rust
```

**Comment by giraffate** [comment]
We've already have the [`dbg_macro`](https://rust-lang.github.io/rust-clippy/master/index.html#dbg_macro) lint.

**Comment by carlomilanesi** [comment]
It is allow-ed. In my opinion, it should be deny-ed by default, or at least warn-ed.

**Comment by matthiaskrgr** [comment]
If you want this to be linted you can always allow the lint in your projects.
I looked at 841 different crate-versions and only 3 of them opted in to have `dbg!()` linted, 42 used the `dbg!()` macro and 799 didn't use `dbg!()`.

So warning about this by default would make 0.3% of the crates "happy" while 95% probably don't care about having `dbg!()` linted and 4.99% would be annoyed, assuming that they don't write `dbg!()` by accident.

Given these numbers I would say the current behavior (opt into the warning) is fine.

**Comment by carlomilanesi** [comment]
OK. I added
#![deny(clippy::dbg_macro)]
as I don't consider good practice to keep dbg calls in production code.
I suspect some of those 799+42 don't know this lint exists, and some of those 42 don't actually need the dbg call in their current code.
